include/swift/Sema/CSBindings.h

```diff
@@ -391,6 +391,13 @@ class BindingSet {
     if (Bindings.empty())
       return false;
 
+    // Literal requirements always result in a subtype/supertype
+    // relationship to a concrete type.
+    if (llvm::any_of(Literals, [](const auto &literal) {
+          return literal.second.viableAsBinding();
```

Comment:
@hborla I have relaxed a check a bit here because non-direct bindings also imply that type variable can't be just a subtype of existential.